includes/internal/emails/class-email-sender.php

```diff
@@ -0,0 +1,276 @@
+<?php
+/**
+ * File containing the Email_Sender class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Emails;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+
+require plugin_dir_path( __DIR__ ) . '../../vendor/autoload.php';
+
+use \Pelago\Emogrifier\CssInliner;
+
+/**
+ * Class Sensei_Email_Sender
+ *
+ * @package Sensei\Internal\Emails
+ */
+class Email_Sender {
+
+	/**
+	 * Email unique identifier meta key.
+	 */
+	public const EMAIL_ID_META_KEY = '_sensei_email_name';
+
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * The constructor.
+	 *
+	 * @var array
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds all filters and actions.
+	 */
+	public function init() {
+		/**
+		 * Send email of predefined types with provided contents.
+		 *
+		 * @param string $email_name   The name of the email template.
+		 * @param array  $replacements The placeholder replacements.
+		 */
+		add_action( 'sensei_send_html_email', [ $this, 'send_email' ], 10, 2 );
+	}
+
+	/**
+	 * Send email of type.
+	 *
+	 * @param string $email_name The email type.
+	 * @param array  $replacements The placeholder replacements.
+	 *
+	 * @access private
+	 */
+	public function send_email( $email_name, $replacements ) {
+		$email_post = $this->get_email_post_by_name( $email_name );
+
+		if ( ! $email_post ) {
```

Comment:
Okay, I see, here is the processing of a missing email post.

Reply:
Yap, that's the plan for now :)